webfrontend/src/components/setup/avatar.tsx

```diff
@@ -0,0 +1,191 @@
+import { createRef, useCallback, useState } from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import AvatarEditor from "react-avatar-editor";
+import { Button, Form, Image } from "react-bootstrap";
+import { triggerErrorMessage } from "../../utils/locals";
+import { postRequestContent } from "../../utils/request";
+import Modal from "react-bootstrap/Modal";
+
+interface ImageProperties {
+  originalImage: string | File;
+  croppedImage: string | undefined;
+  position: { x: number; y: number };
+  scale: number;
+  rotate: number;
+}
+
+interface Props {
+  currentProfileImg: string | File;
+  setIsEditProfileImg: React.Dispatch<React.SetStateAction<boolean>>;
+  isEditProfileImg: boolean;
+  setProfileImgUrl: React.Dispatch<React.SetStateAction<File | undefined>>;
+}
+
+const Avatar: React.FC<Props> = ({
+  setIsEditProfileImg,
+  isEditProfileImg,
+  setProfileImgUrl,
+  currentProfileImg,
+}) => {
+  const editorRef: React.RefObject<AvatarEditor> = createRef();
+  const [imgToBeSubmitted, setImgToBeSubmitted] = useState<File>();
+  const [fileName, setFileName] = useState("");
+  const [imageProperties, setImageProperties] = useState<ImageProperties>({
+    originalImage: currentProfileImg,
+    croppedImage: undefined,
+    position: { x: 0.5, y: 0.5 },
+    scale: 1,
+    rotate: 0,
+  });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const { originalImage, croppedImage, position, scale, rotate } =
+    imageProperties;
```

Comment:
I dont understand the point of using a dictionary as a state and then declare derived states, this would make your code harder to debug. Either only use the dictionary state, and read directly from that dictionary, or create individual states.